src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -130,19 +132,17 @@ internal IRuntimeClient RuntimeClient
         #endregion
 
         #region Static initialization
-        
-        public SerializationManager(IServiceProvider serviceProvider, IMessagingConfiguration config, ITraceConfiguration traceConfig, ILoggerFactory loggerFactory)
+
+        public SerializationManager(IServiceProvider serviceProvider, IOptions<SerializationProviderOptions> serializatonProviderOptions, ITraceConfiguration traceConfig, ILoggerFactory loggerFactory)
         {
-            var serializationProviders = config.SerializationProviders;
-            var fallbackType = config.FallbackSerializationProvider;
             this.LargeObjectSizeThreshold = Constants.LARGE_OBJECT_HEAP_THRESHOLD;
             this.serializationContext = new ThreadLocal<SerializationContext>(() => new SerializationContext(this));
             this.deserializationContext = new ThreadLocal<DeserializationContext>(() => new DeserializationContext(this));
 
             logger = new LoggerWrapper<SerializationManager>(loggerFactory);
             this.serviceProvider = serviceProvider;
             RegisterBuiltInSerializers();
-            fallbackSerializer = GetFallbackSerializer(serviceProvider, fallbackType);
+            fallbackSerializer = GetFallbackSerializer(serviceProvider, serializatonProviderOptions.Value.FallbackSerializationProvider);
```

Comment:
similar, just access the .Value once and then use that unwrapped object as many times as you want